### PR TITLE
Move to a paginated layout

### LINKED ISSUE #2086: Implement paginated view
See #2062 for prototype

### This task is done when...
- [x] sidebar is updated to open each section in a page
- [x] sidebar is updated to only expand the current section
- [x] update the sidebar to make sure the current section is highlighted both when the page loads and as the user scrolls
- [x] scroll to the top when switching pages (ensure that keyboard focus management works properly, too)

### PR DESCRIPTION
**Pull request by mgwalker**
### This pull request changes...

- does what it says! 🎉
- closes #2086 

### This pull request is ready to merge when...

- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- N/A ~The change has been documented~
- [ ] Changelog is updated as appropriate

### This feature is done when...

- [ ] Design has approved the experience
- [x] Product has approved the experience

### CODE REVIEW

File: web/src/containers/Sidebar.test.js [hunk 1]
```diff
@@ -9,32 +8,111 @@ import {
...
+});
+
+global.scrollTo = jest.fn();
+
 describe('Sidebar component', () => {
   const props = {
     activities: [
-      { anchor: '#key1', key: 'key 1' },
-      { anchor: '#key2', key: 'key 2' }
+      { anchor: '#key1', key: 'key 1234' },
```
Comment (mgwalker) [inline web/src/containers/Sidebar.test.js]: Why this change? Because keys are 8-characters long, and the sidebar now relies on key length to figure out if an activity subsection is selected. So the test keys need to be 8 characters, too.

File: web/src/styles/scss/section.scss [hunk 2]
```diff
@@ -1,42 +1,4 @@
...
-
-  margin-top: $spacer-6;
-  padding-top: $spacer-2;
-  padding-left: $number-padding-size;
-  border-top: 1px solid $color-gray-dark;
-
-  &:first-of-type {
-    margin-top: $spacer-2;
-  }
-}
```
Comment (mgwalker) [inline web/src/styles/scss/section.scss]: @hbillings I think getting rid of all this is fine since we don't have numbered sections anymore. Is that right? Is that what the `counter-reset` bit is for too?

**Comment by lauraponce** [comment]
@mgwalker the test link seems to be hanging and timing out. Is there a way to reset it or anything?

**Comment by mgwalker** [comment]
@lauraponce I'll re-run the build so it'll redeploy. I'll also delete the old bot comment so we'll know for sure when it's done redeploying.

**Comment by mgwalker** [comment]
big oops

**Comment by CSwartzHMA** [comment]
This all looks great and appears to function as described.
